tests/compatibility-test.py

```diff
@@ -263,106 +267,36 @@ def get_new_value():
-        # replacement. Finally, we will try to connect to the detached actor again.
         client = docker.from_env()
         container = client.containers.run(ray_image, remove=True, detach=True, stdin_open=True, tty=True,
-                                          network_mode='host', command=["/bin/sh", "-c", "python"])
-        s = container.attach_socket(
-            params={'stdin': 1, 'stream': 1, 'stdout': 1, 'stderr': 1})
-        s._sock.setblocking(0)
-        s._sock.sendall(b'''
-import ray
-import time
+                                            network_mode='host', command=["/bin/sh"])
+        ray_namespace = ''.join(random.choices(string.ascii_lowercase, k=10))
+        logger.info(f'namespace: {ray_namespace}')
 
-def retry_with_timeout(func, count=90):
-    tmp = 0
-    err = None
-    while tmp < count:
-        try:
-            return func()
-        except Exception as e:
-            err = e
-            tmp += 1
-    assert err is not None
-    raise err
+        # Register a detached actor
+        utils.copy_to_container(container, 'tests/scripts', '/usr/local/', 'test_detached_actor_1.py')
+        exit_code, output = utils.exec_run_container(container, f'python3 /usr/local/test_detached_actor_1.py {ray_namespace}', 180)
 
-ray.init(address='ray://127.0.0.1:10001')
-
-@ray.remote
-class A:
-    def ready(self):
-        import os
-        return os.getpid()
-
-a = A.options(name="a", lifetime="detached", max_restarts=-1).remote()
-res1 = ray.get(a.ready.remote())
-print('ready')
-
-        ''')
-
-        count = 0
-        while count < 90:
-            try:
-                buf = s._sock.recv(4096)
-                logger.info(buf.decode())
-                if buf.decode().find('ready') != -1:
-                    break
-            except Exception as e:
-                pass
-            time.sleep(1)
-            count += 1
-        if count >= 90:
-            raise Exception('failed to run script')
+        if exit_code != 0:
+            raise Exception(f"Exception happens during the execution of test_detached_actor_1.py. The exit code is {exit_code}.")
 
-        # kill the gcs on head node. If fate sharing is enabled
-        # the whole head node pod will terminate.
+        # Kill the gcs on head node. If fate sharing is enabled, the whole head node pod will terminate.
         utils.shell_assert_success(
             'kubectl exec -it $(kubectl get pods -A| grep -e "-head" | awk "{print \\$2}") -- /bin/bash -c "ps aux | grep gcs_server | grep -v grep | awk \'{print \$2}\' | xargs kill"')
-        # wait for new head node getting created
-        time.sleep(10)
-        # make sure the new head is ready
-        utils.shell_assert_success(
-            'kubectl wait --for=condition=Ready pod/$(kubectl get pods -A | grep -e "-head" | awk "{print \$2}") --timeout=900s')
+        # Wait for new head node getting created
+        # TODO (kevin85421): Need a better method to wait for the new head pod.
```

Comment:
I am obliged by the laws of code review to ask:
can we solve this TODO in this PR?

I understand that the process by which the head restarts seems a little random and unpredictable right now,
but do you think there's a sequence of events that happens consistently that we could validate here?
()

Reply:
We can also solve this in this PR. This issue is a part of https://github.com/ray-project/kuberay/issues/618.

Reply:
Got it -- no need to solve in this PR then. Could you link the issue in the TODO?

Reply:
Added the link.